src/webapp/components/landing-page-edit-dialog/LandingPageEditDialog.tsx

```diff
@@ -202,8 +200,49 @@ export const LandingPageEditDialog: React.FC<LandingPageEditDialogProps> = props
                         <p>{i18n.t("Bottom")}</p>
                     </IconLocationSwitch>
                 </div>
+
+                <Label>{i18n.t("Icon Size")}</Label>
+                <OptionContainer>
+                    {["small", "medium", "large"].map((size, i) => (
+                        <Button
+                            key={i}
+                            color={
+                                size === value.iconSize || (!value.iconSize && size === "small") ? "primary" : "default"
+                            }
+                            variant="contained"
+                            value={size}
+                            onClick={() => onChangeIconSize(size)}
```

Comment:
**[No need to change]** In this case doesn't apply because these three buttons will re-render anyway because of `[].map()`. But in other case, these three buttons would also re-render each time because of the creation of the arrow function.

```
(()=>{}) === (()=>{}) // output: false
```

React will treat the props as different and would cause the button to re-render each time. And same applies for arrays and objects:

```
[] === [] // false
({}) == ({}) // false
Object() === Object() // false
```

A more meaningful example would be:

```
["small", "medium", "large"] === ["small", "medium", "large"] // output: false
```

That's why these buttons will be re-rendered each time. A common solution would be to declare the array outside of the component scope (as you know we usually do it at the end of the file along other constants). And also adding the `() =>` part to the `useCallback`.

This was a heavy headache for me to understand... and it took me 2 years to realize 🫠. This doesn't apply to conditions where the value is a string or a number because: `"hello" === "hello" // true`. That's why is not necessary for the color prop. It also took me a lot of reviews from Arnau and for sure if nobody reviews my code I will not do it perfectly on each given prop :')

Another thing to consider here is why I put **[No need to change]**; this button from Material UI is straightforward and I'm sure it will not cost much effort to re-render. But this is really a MUST if the component is a custom one with complex functionality, for example, this component itself is a more complex one `LandingPageEditDialog` than a button. So don't change it but be aware of it 🥷🏼.

Reply:
got it. thanks @p3rcypj 😀